### PR TITLE
GitHub Authentication

### LINKED ISSUE #97: GitHub Access
The Version Control tutorial doesn't cover how to authenticate so that you can *write* to a repo.  For that, you need either SSH keys or a GitHub Access token.  I think that most students will find Access Tokens easier.

In office hours today, I had a student run into this problem.  Furthermore, she needed to use the "classic token" to get it to work, not the "fine-grain token".

### PR DESCRIPTION
**Pull request by TheOmid**
Add GitHub Authentication section that walks through SSH Keys and GitHub Personal Access Tokens.

A couple of notes:

1. Maybe we should just include a tutorial for downloading [Git Credential Manager](https://docs.github.com/en/enterprise-server@3.4/get-started/getting-started-with-git/updating-credentials-from-the-macos-keychain#updating-your-credentials-via-keychain-access) which takes care of all of it. It makes it much easier for students IMO.
2. I think we should tell students to set an expiration date for their password. Having no expiration is a security risk. The only concern is that it can create a hassle for them to renew it. 90 days is also not enough. Ideally, we can ask students to set it as the last day of classes, but for now, I have put it down January 1st of next year. My main concern here is that how to update your access token is confusing. I think I should write it as a subsection.
3. I think we should update the whole tutorial and center it around pushing project 2 to GitHub. Since this is out of the scope of this issue, I did not change other areas besides connecting to remote repo. Student's might get confused that the rest of the project is for Justin's EECS 485 project 1.

I think the MacOS pitfall and expiration note can be more clear. LMK if you have suggestions for that
I personally use SSH. I tested this with a new GitHub account, and the tutorial worked.

Closes #97

### CODE REVIEW

**Review by awdeorio — COMMENTED** [review]
This is great!  I pushed a lot of changes in response to your thoughts.  Please take a look and let me know what you think.

> Maybe we should just include a tutorial for downloading Git Credential Manager which takes care of all of it. It makes it much easier for students IMO.

I'm hesitant to get more tools involved.  I think I'd lean more towards recommending SSH keys instead.  I tried that in my edits, take a look.

> I think we should tell students to set an expiration date for their password. Having no expiration is a security risk. The only concern is that it can create a hassle for them to renew it. 90 days is also not enough. Ideally, we can ask students to set it as the last day of classes, but for now, I have put it down January 1st of next year. My main concern here is that how to update your access token is confusing. I think I should write it as a subsection.

I tried a longer date, but I think one year is the max.

> I think we should update the whole tutorial and center it around pushing project 2 to GitHub. Since this is out of the scope of this issue, I did not change other areas besides connecting to remote repo. Student's might get confused that the rest of the project is for Justin's EECS 485 project 1.

I agree.  I made an issue and assigned you.  #101

> I think the MacOS pitfall and expiration note can be more clear. LMK if you have suggestions for that

I don't quite understand.

> I personally use SSH. I tested this with a new GitHub account, and the tutorial worked.

I agree.  I moved SSH keys to "recommended".  Take a look.

File: docs/setup_git.md [hunk 1]
```diff
@@ -198,6 +198,60 @@ nothing to commit, working tree clean
...
+
+Avoid entering your access token every time by configuring the local Git client to save the GitHub personal access token. 
+```console
+$ git config --global credential.helper store
+```
+
+#### Copy token
+{: .primer-spec-toc-ignore }
+
+Copy your token.  You'll need it later when you do `git push`.  **Do not close the browser tab.**
```
Comment (TheOmid) [inline docs/setup_git.md:251]: I think we can again emphasize this your only chance to copy it

**Comment by TheOmid** [comment]
The edits look good; thanks for making it more readable!

I have a couple of comments:

1- Since SSH is not super beginner friendly, that's why we decided to lean towards it. I think it would be good to make an SSH section instead of relying on GitHub (although in my opinion, it does a great job students told me they were confused). I can work on this! 

2- You have removed hooks and delete_repo using a personal access token in the personal access token. Can they delete their repository without the token? (Is it to delete with an API call?). If the other two are API specific, we can ignore it, but it was also in GitHub's tutorial. Hence I included it.

File: docs/setup_git.md [hunk 2]
```diff
@@ -198,6 +198,60 @@ nothing to commit, working tree clean
 ```
 </div>
 
+## GitHub Authentication
+There are two ways to connect to GitHub: SSH Keys and GitHub Personal Access Tokens Keys.  An SSH Key is a special file that you can use to connect to remote terminals.  A Personal Access Token works like a separate password used just for GitHub.
+
+### SSH Keys (recommended)
+Follow the GitHub [Connecting to GitHub with SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh) tutorial.
```
Comment (TheOmid) [inline docs/setup_git.md:205]: Students might think they need to go through all the links, we can ask them to go through [Generating a new SSH key and adding it to the ssh-agent](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).

**Review by awdeorio — APPROVED** [review]
@anishnya and I team coded this and reviewed it together.  It's good to go.

**Comment by TheOmid** [comment]
Only thing I forgot to mention is that here:

"MacOS Pitfall: If you are not prompted for your username and password, your credentials may be cached on your computer. [Clear your credentials from Keychain Access](https://docs.github.com/en/enterprise-server@3.4/get-started/getting-started-with-git/updating-credentials-from-the-macos-keychain#updating-your-credentials-via-keychain-access)."

Do we probably want to say this if to Personal Access Token? 

Also, maybe mention that with SSH you won't see a prompt for entering a password? I don't see SSH being mentioned as password-less authentication.

**Comment by awdeorio** [comment]
Good catch.  Could you make a separate PR with these proposed changes @TheOmid ?

**Comment by TheOmid** [comment]
Will do!

**Pull request merged**